File: Programmers/LEVEL_1/12906-같은_숫자는_싫어.js
/**
* ABOUT
*
* DATE: 2021-03-29
* AUTHOR: lovelysi0113
*
* URL: https://programmers.co.kr/learn/courses/30/lessons/12906
*
* COMMENT:
* 연속적으로 나타나지 않을 때만 정답 배열에 넣어준다
**/

function solution(arr)
{
    var answer = [];
    let lastIdx = -1;

    for (let idx = 0; idx < arr.length; idx++) {
        if (arr[idx] !== answer[lastIdx]) {
            answer.push(arr[idx]);
            lastIdx++;
        }
    }
    
    return answer;
}

File: Programmers/LEVEL_1/12906-같은_숫자는_싫어.py
'''/**
* ABOUT
*
* DATE: 2021-03-29
* AUTHOR: lovelysi0113
*
* URL: https://programmers.co.kr/learn/courses/30/lessons/12906
*
* COMMENT:
* 연속적으로 나타나지 않을 때만 정답 배열에 넣어준다
**/'''

def solution(arr):
    answer = []
    temp = -1
    for chk in arr:
        if temp != chk:
            temp = chk
            answer.append(temp)
    return answer